### ISSUE TITLE
reduction removes notes and stems in case of mixed normal and small heads

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#364]
![chandelier](https://user-images.githubusercontent.com/33522630/76104381-8a39bf80-5fd3-11ea-8b97-33149766471a.jpg)

In this score there are a couple of chords that use a normal and a small head.
Unfortunately most of the - already detected - stems and heads are removed in step reduction.
I also tried to acitvate "detect small heads" - but no change.

**Comment by MusicForStringsPercussionCeleste** [comment]
I would not worry about capturing those small note heads. Basses routinely double octaves without needing the notes. Plus those are useless notes unless singing with chorus of  Finnish Monks 😁
In vocal music, besides bass doubling or a possible baritone note, small notes are used for alternate melodies in songs with multiple verses.  Maybe someone can attempt scanning a strophic song like that.

(Please excuse autocorrect)
Richard

On Mar 6, 2020, at 11:57 AM, Baruch <notifications@github.com> wrote:

﻿

[chandelier]<https://user-images.githubusercontent.com/33522630/76104381-8a39bf80-5fd3-11ea-8b97-33149766471a.jpg>

In this score there are a couple of chords that use a normal and a small head.
Unfortunately most of the - already detected - stems and heads are removed in step reduction.
I also tried to acitvate "detect small heads" - but no change.

**Comment by Bacchushlg** [comment]
The small notes alone don't bother me. But the notes are ignored completely...

**Comment by MusicForStringsPercussionCeleste** [comment]
yes, but they have low value and a tiny % of appearance in music; near Zero appearance in instrumental music.
Pick your battles, go for high-value targets.

________________________________

**Comment by hbitteur** [comment]
In Audiveris OMR engine, there is an explicit exclusion between small and standard items on the same chord.
You can't have a small head and a standard head on the same stem.
Also, standard heads support connected standard beams (and small heads support connected small beams), though they are not directly connected but via a stem.

Here, look at head grades before reduction: the small heads have lower grades than the standard heads on the common stem, hence they get removed because of the exclusion rule.
But the resulting stem must have a tail side and a head side, here it now has just heads in the middle of the stem. The connection between remaining standard head and stem is thus removed.

And the standard head is now alone, with no stem connected. It is thus removed.

**Comment by Bacchushlg** [comment]
You should think over this reduction rule - in choir scores you find this type of notation rather often. The small notes are meant as "optional" for a part the singers of the voice.
Is there any other reason to insist on this rule?

**Comment by hbitteur** [comment]
Many small heads are badly recognized, so it was a way to discard them. I was used to see small heads on their own, not mixed with standard heads on the same stem. Obviously I was wrong, I went too far when I added this exclusion rule.

I will check if we can rather safely remove it.
